lib/carrierwave/storage/postgresql_lo.rb

```diff
@@ -71,11 +71,12 @@ def store!(file)
 
       def retrieve!(identifier)
         raise "This uploader must be mounted in an ActiveRecord model to work" unless uploader.model
+        @oid = identifier
         CarrierWave::Storage::PostgresqlLo::File.new(uploader)
       end
 
       def identifier
-        @oid ||= uploader.model.read_attribute(uploader.mounted_as) || connection.lo_creat
+        @oid ||= connection.lo_creat
```

Comment:
It seems to me that with this patch a new kid is always created, regardless of what is stored in the database